controllers/configmap_controller.go

```diff
@@ -181,8 +183,11 @@ func (r *ConfigMapReconciler) reconcileNodes(ctx context.Context, instances *cor
 			return errors.Wrapf(err, "error configuring host with address %s", host.Address)
 		}
 	}
-	// TODO: Remove BYOH nodes that are not associated with the parsed host list
-	//       https://issues.redhat.com/browse/WINC-582
+
+	// Ensure that only instances specified by the ConfigMap are joined to the cluster as nodes
```

Comment:
removeUndesiredNodes seems confusing to me, having a specific comment like instances *currrently specified by ConfigMap might  be more clear.
Would it make sense to name this function to something like syncNodesToConfigMap? 
On a similar note, might it be worth considering a flow where we sync the configmap instances with current list of nodes -> decide to remove or add nodes.